core/src/apps/cardano/sign_tx.py

```diff
@@ -125,117 +109,82 @@ async def sign_tx(ctx, msg, keychain: seed.Keychain):
 class Transaction:
     def __init__(
         self,
-        inputs: list,
-        outputs: list,
-        keychain,
+        inputs: List[CardanoTxInputType],
+        outputs: List[CardanoTxOutputType],
+        keychain: seed.Keychain,
         protocol_magic: int,
-        input_coins_sum: int,
-    ):
+        fee: int,
+        ttl: int,
+    ) -> None:
         self.inputs = inputs
         self.outputs = outputs
         self.keychain = keychain
```

Comment:
`keychain` is a different type of argument than the rest, so I would move it to the front of the list.

Reply:
I agree, moved.